core/src/main/java/org/springframework/security/provisioning/JdbcUserDetailsManager.java

```diff
@@ -178,7 +186,7 @@ protected void initDao() throws ApplicationContextException {
 	 */
 	@Override
 	protected List<UserDetails> loadUsersByUsername(String username) {
-		return getJdbcTemplate().query(getUsersByUsernameQuery(), this::mapToUser, username);
+		return getJdbcTemplate().query(getUsersByUsernameQuery(), userDetailsMapper, username);
```

Comment:
Please also add two unit tests:

1. One that confirms that setting a `null` value for `setUserDetailsMapper` will throw an exception
2. One that confirms that setting a mock value gets used when querying (call `setUserDetailMapper` with a Mockito mock, search for a user, and verify that the mock row mapper was used)